src/test/java/org/codehaus/plexus/util/xml/Xpp3DomWriterTest.java

```diff
@@ -82,6 +82,8 @@ private String createExpectedXML(boolean escape) {
         buf.append(LS);
         buf.append("  </el6>");
         buf.append(LS);
+        buf.append("  <el8>special-char-" + (char) 7 + "</el8>");
```

Comment:
@michael-o @elharo
special char here should be removed or replaced .... escaping is not enough
it is forrbiden char in xml ....

Reply:
I'd like to have a reference for that. I believe that you are right, but I also remember that entities are permissive.

Reply:
See section 2.2 and 4.1 of the XML 1,.0 spec: https://www.w3.org/TR/REC-xml/#charsets
"Characters referred to using character references MUST match the production for [Char](https://www.w3.org/TR/REC-xml/#NT-Char)."

XML 1,1 is a little more permissive, but still prohibits the NUL character.

Reply:
Yes, noticed as well. My approach is not correct.